### DISCUSSION TITLE
100% CPU usage after launching server

### DISCUSSION THREAD

**Discussion by nayanw775** [PlusToolkit/PlusLib#790] [Q&A] [answered]
Hi PlusToolkit Team,

We are using PlusApp and slicer for the navigation purpose, we have successfully created a server using OpenIGTLinkIF and PlusApp. So issue I am facing is my system gets very slow after connecting OpenIGTLinkIF and PlusApp and this exactly happens when I press Active button on the OpenIGTLink module after creating the connector.
So before launching the server I have noticed CPU usage is somewhere between 5-10%  before pressing the Active button and as soon as I make connection active and Start tracking it goes to100%.
Previously we were using PC with i3 2GB graphic card, and 8GB RAM so we thought PC configuration may be making this lapse then we changed it to:

PC configuration - OS -Windows 10, RAM -16GB, Graphic Card-6GB Nvidia, Core i7 9th Generation.

I suppose PC configuration is fine, may be I need to change some settings in the PlusApp or IGTLinkIF, I don’t know exactly where the problem lies.
What should I change in the code to reduce my CPU usage and to minimize the lapse or should I try changing my PC settings from somewhere.
PlusApp Version:2.8.0
Slicer Version: 4.11

[Config.txt](https://github.com/PlusToolkit/PlusLib/files/6014154/Config.txt)

**Comment by adamrankin** [comment]
Are all your cores at 100%?

**Reply by nayanw775** [reply]
Yes my all cores are at 100% usage

**Comment by Sunderlandkyl** [comment]
If you set EnableCapturingOnStart to "FALSE" in the capture device, does it reduce the CPU usage?

**Reply by nayanw775** [reply]
Thanks for the quick response I have to check that, I will set it to false and will get back to you.

**Comment by adamrankin** [comment]
I do see that enable compression is on, but it seems excessive that a single tracker device would use 100%. Depends if’s all cores or a single core.

Adam

**Reply by nayanw775** [reply]
Machine have total of 6cores and all are at 100% usage

**Reply by nayanw775** [reply]
Hi  @adamrankin  and @Sunderlandkyl 

Sorry for writing so late, I went ahead with the changes that you suggested, CPU usage has reduced just a bit, I do not what configurations others are using, because I have read people using the dual core CPU also for the same process and still manage to run smoothly.

Would you like to suggest any other changes.


Thanks,
Nayan

**Comment by nayanw775** [comment]
So would you suggest changes that you have mentioned in this thread
[https://discourse.slicer.org/t/high-slicer-cpu-usage-while-live-streaming/6078](url)

**Comment by nayanw775** [comment]
![cpu usage](https://user-images.githubusercontent.com/72604275/110598445-c33cdf00-81a7-11eb-8eda-68afc157886d.png)
![cpu usage2](https://user-images.githubusercontent.com/72604275/110598457-c768fc80-81a7-11eb-8761-6641670a93c3.png)


So here are the screenshots of the CPU usage before and after launching the Server

**Reply by adamrankin** [reply]
Can you post your config file?

**Reply by nayanw775** [reply]
[Config.txt](https://github.com/PlusToolkit/PlusLib/files/6120263/Config.txt)

**Reply by nayanw775** [reply]
Hi @adamrankin Any updates on this, And yes I do have made changes to the config file as suggested @Sunderlandkyl and this usage is after the changes.

**Reply by adamrankin** [reply]
I've replicated your setup and I am not experiencing the increase. I am continuing to investigate.

**Reply by nayanw775** [reply]
Hi @adamrankin thanks for the update one more thing I would like to confirm, while testing are you using Volume rendering or models for the registration(if only you are doing)

**Reply by adamrankin** [reply]
No, I'm only streaming data from Plus to Slicer. In your screenshots, is it PlusServer.exe that is at 100%, or SlicerApp-Real.exe?

**Reply by nayanw775** [reply]
It is SlicerApp-Real.exe is 100%
But it only turns to 100% once I launch the server

**Reply by adamrankin** [reply]
Ah, this isn't a bug in Plus. I suspect your volume rendering is set to CPU render. Check the volume rendering panel for CPU/GPU options.

**Comment by adamrankin** [comment, marked as answer]
https://github.com/PlusToolkit/PlusLib/discussions/790#discussioncomment-513128

Not a bug, CPU usage is from Slicer.

**Reply by lassoan** [reply]
You can reduce CPU usage in Slicer by switching to GPU volume rendering.
